src/machine/operation.c

```diff
@@ -8,8 +8,33 @@
 #include "operation.h"
 #include "process-util.h"
 
+static int operation_done_internal(const siginfo_t *si, Operation *o, void *userdata) {
+        int r;
+
+        assert(si);
+        assert(o);
+
+        if (si->si_code != CLD_EXITED)
+                return log_debug_errno(SYNTHETIC_ERRNO(ESHUTDOWN), "Child died abnormally");
+
+        if (si->si_status == EXIT_SUCCESS)
+                r = 0;
+        else if (read(o->errno_fd, &r, sizeof(r)) != sizeof(r)) /* Try to acquire error code for failed operation */
+                return log_debug_errno(errno, "Failed to read failed operation's errno: %m");
```

Comment:
```c
else {
        ssize_t n = read(...);
        if (n < 0)
                return log_debug_errno(errno, "...");
        if (n != sizeof(r))
                return log_debug_errno(SYNTHETIC_ERRNO(EIO), "...");
}
```

Reply:
Not sure, but in this case, propagating such errors is OK, I think.
Do you have any good idea to handle such errors gracefully?

Reply:
usually I handle it like this:

```
        for (;;) {
                if (write(fd, message, length) >= 0)
                        return 0;
                if (errno == EINTR)
                        continue;
                if (errno != EAGAIN)
                        return -errno;
         }
```

There is also a variant when one can count sent/read bytes because an operation maybe partial. Which is especially true for network operation.

The code above I copy-pasted from [resolved code](https://github.com/systemd/systemd/blob/main/src/resolve/resolved-manager.c#L1002)